R/process/01-process-neonSoilFlux.R

```diff
@@ -129,40 +136,63 @@ env_values <- vector(mode = "list", length = length(env_files))
-    select(-surface_diffusivity,-flux_compute) |>
-    unnest(cols=c(joined_data)) |>
+    mutate(joined_data = map2(.x = flux_compute,
+                              .y = surface_diffusivity,
+                              .f = ~ left_join(.x,
+                                               .y,
+                                               by = "diffus_method"))) |>
+    select(-surface_diffusivity, -flux_compute) |>
+    unnest(cols = c(joined_data)) |>
     group_by(diffus_method) |>
     nest() |>
-    mutate(data = map(data,.f=~(.x |>
-                                  select(-gradient,-gradient_err) |>
-                                  group_by(startDateTime,horizontalPosition,soilCO2concentrationMeanQF,VSWCMeanQF,soilTempMeanQF,staPresMeanQF,diffusivity,diffusExpUncert,zOffset) |>
-                                  nest() |>
-                                  rename(flux_compute = data) |>
-                                  ungroup() |>
-                                  group_by(startDateTime,horizontalPosition,flux_compute,soilCO2concentrationMeanQF,VSWCMeanQF,soilTempMeanQF,staPresMeanQF) |>
-                                  nest() |>
-                                  rename(diffusivity = data) |>
-                                  ungroup() |>
-                                  relocate(startDateTime,horizontalPosition,flux_compute,diffusivity,soilCO2concentrationMeanQF,VSWCMeanQF,soilTempMeanQF,staPresMeanQF)
-
+    mutate(data = map(data, .f = ~ (.x |>
+      select(-gradient, -gradient_err) |>
+      group_by(startDateTime,
+               horizontalPosition,
+               soilCO2concentrationMeanQF,
+               VSWCMeanQF, soilTempMeanQF,
+               staPresMeanQF,
+               diffusivity,
+               diffusExpUncert,
+               zOffset) |>
+      nest() |>
+      rename(flux_compute = data) |>
+      ungroup() |>
+      group_by(startDateTime,
+               horizontalPosition,
+               flux_compute,
+               soilCO2concentrationMeanQF,
+               VSWCMeanQF,
+               soilTempMeanQF,
+               staPresMeanQF) |>
+      nest() |>
+      rename(diffusivity = data) |>
+      ungroup() |>
+      relocate(startDateTime,
+               horizontalPosition,
+               flux_compute, diffusivity,
+               soilCO2concentrationMeanQF,
+               VSWCMeanQF,
+               soilTempMeanQF,
+               staPresMeanQF)
     )))
 
   # Assign each of the fluxes to a different data frame
   model_fluxes_mq[[i]] <- nested_flux_diffus |>
-    filter(diffus_method=="Millington-Quirk") |>
+    filter(diffus_method == "Millington-Quirk") |>
     pull(data) |>
+    pluck(1) |> # return df not list of length 1
     mutate(site = site_name)
 
   model_fluxes_marshall[[i]] <- nested_flux_diffus |>
-    filter(diffus_method=="Marshall") |>
+    filter(diffus_method == "Marshall") |>
     pull(data) |>
+    pluck(1) |> # return df not list of length 1
```

Comment:
Had to add this to get data frame out instead of list of length 1 with data frame in the first element